src/index.js

```diff
@@ -200,3 +200,50 @@ Hooks.on('canvasInit', () => {
 });
 
 Hooks.on('renderChatMessage', (_, html) => Item5e.chatListeners(html));
+
+Hooks.on("createToken", async (token) => {
+  console.log(token.actor.type)
+  // Checks if its a NPC type of actor and if the game setting is set to true
+  if (token.actor.type === "npc" && game.settings.get("a5e", "npcHealthRandomization")) {
+    console.log("made it!")
+    // getting the NPC Hit Dice
+    const hitDice = token.actor.system.attributes.hitDice;
+    
+    // Get Constitution Modifier
+    const conMod = token.actor.system.abilities.con.mod;
+
+    // Set variable to track total number of hitDice for con mod multiplier later
+    let hitDiceTotalCount = 0;
+  
+    // Build Hit Dice formula for Roll
+    let hitDiceFormula = "";
+    for (const prop in hitDice){
+      if(hitDice[prop].total > 0){
+        if (hitDiceFormula !== "") {
+          hitDiceFormula += "+";
+        } 
+        // Add hit dice to formula
+        hitDiceFormula += hitDice[prop].total + prop;
+        // Increment Hit Dice Count total
+        hitDiceTotalCount++;
+      }
+    }
+
+    if (hitDiceTotalCount > 0) {
```

Comment:
It's always better to guard clauses.

In this case that would be `if (hitDiceTotalCount == 0) return`

Reply:
`===` 😉